package/MDAnalysis/analysis/base.py

```diff
@@ -119,6 +123,12 @@ def __delattr__(self, attr):
             raise AttributeError("'Results' object has no "
                                  f"attribute '{attr}'") from err
 
+    def __getstate__(self):
+        return self.data
+
+    def __setstate__(self, state):
+        self.data = state
+
```

Comment:
Is this part not handled by UserDict?

Reply:
I think `UserDict` itself can be properly pickled, but when its `__getattr__` is overridden, `__getstate__` and `__setstate__` have to be defined. 
https://stackoverflow.com/questions/50888391/pickle-of-object-with-getattr-method-in-python-returns-typeerror-object-no